darts/timeseries.py

```diff
@@ -880,6 +880,41 @@ def is_within_range(self,
         index = self.time_index()
         return index[0] <= ts <= index[-1]
 
+
+    def map(self,
+            fn: Callable[[float], float],
+            cols: Optional[Union[List[int], int]] = None) -> 'TimeSeries':
+        #TODO check if typing is correct
+        """
+        Applies the function fn to all values in this TimeSeries, or, to only those
+        values in the columns specified by the optional argument cols. Returns a new
+        TimeSeries instance.
+        
+        Parameters
+        ----------
+        fn
+            A numerical function
+        cols
+            Optionally, an integer or list of integers specifying the column(s) onto which fn should be applied
+
+        Returns
+        -------
+        TimeSeries
+            A new TimeSeries instance
+        """
+        new_dataframe = self.pd_dataframe() 
+        
+        if cols is None:
+            new_dataframe = new_dataframe.applymap(fn)
```

Comment:
In this case (when `cols is None`), you don't need the (expensive) call to `self.pd_dataframe()` on line 905 because `applymap()` already returns a new DF instance. I would call `self.pd_dataframe()` only just before line 915.

Reply:
Done.